src/components/Pagination.js

```diff
@@ -14,6 +20,40 @@ const Pagination = ({ pageContext }) => {
             </Link>
           </div>
         )}
+        {Array.from({ length: numberOfPages }, (_, i) => {
+          const index = i + 1
+          const renderLink =
+            index <= endSize ||
+            (index >= humanPageNumber - midSize &&
+              index <= humanPageNumber + midSize) ||
+            index > numberOfPages - endSize
+          let NavItem = null
+          if (index === humanPageNumber) {
+            dots = true
+            NavItem = (
+              <div className="navbar-item" key={`page-${index}`}>
+                <span aria-current="page" className="current">
+                  {index}
+                </span>
+              </div>
+            )
+          } else if (renderLink) {
+            dots = true
+            NavItem = (
+              <div className="navbar-item" key={`page-${index}`}>
+                <Link to={i === 0 ? `/` : `/${index}`}>{index}</Link>
```

Comment:
I’ve been thinking a bit about this one. We have a pagination helper in the awesome pagination plugin. The trick is, we also support defining a function to calculate what the paginated paths should be. In that case, there’s no way to calculate the page number links without that function.

I’m thinking maybe we add some logic that allows for that function to be passed to the pagination component, with the same default used in both the `paginate()` function and the component. It’s an interesting one. Will give it a little more thought...